src/createStore.ts

```diff
@@ -186,15 +217,15 @@ export default function createStore<
 
     let isSubscribed = true
 
-    ensureCanMutateNextListeners()
+    ensureCanMutateNextListeners() // method defined above to make sure nextListeners can be updated
     nextListeners.push(listener)
 
     return function unsubscribe() {
       if (!isSubscribed) {
         return
       }
 
-      if (isDispatching) {
+      if (isDispatching) { // more checks
```

Comment:
I like seeing how all of this detailed error handling works. I'm not sure about the rest of you, but I don't handle errors at this level in my day-to-day.

Reply:
For sure! I am sure all those downloads with tons of bugs and issues have brought them to this point.